src/components/Explore/TracesByService/Tabs/Structure/StructureScene.tsx

```diff
@@ -55,34 +60,88 @@ export class StructureTabScene extends SceneObjectBase<ServicesTabSceneState> {
       ...state,
     });
 
-    this.addActivationHandler(this._onActivate.bind(this));
+    this.addActivationHandler(() => {
+      this._onActivate();
+      return () => this._onDeactivate();
+    });
+  }
+
+  private _onDeactivate() {
+    if (this.updateTimeoutId) {
+      clearTimeout(this.updateTimeoutId);
+      this.updateTimeoutId = null;
+    }
+    this.lastProcessedData = null;
   }
 
   public _onActivate() {
     this.state.$data?.subscribeToState((state) => {
-      this.setState({ loading: state.data?.state === LoadingState.Loading });
-
+      if (state.data?.state === LoadingState.Loading) {
+        this.setState({ loading: true });
+        return;
+      }
+      
       if (
         (state.data?.state === LoadingState.Done || state.data?.state === LoadingState.Streaming) &&
         state.data?.series.length
       ) {
         const frame = state.data?.series[0].fields[0].values[0];
         if (frame) {
-          const response = JSON.parse(frame) as TraceSearchMetadata[];
-          const merged = mergeTraces(response);
-          merged.children.sort((a, b) => countSpans(b) - countSpans(a));
-          this.setState({
-            tree: merged,
-            panel: new SceneFlexLayout({
-              height: '100%',
-              wrap: 'wrap',
-              children: this.getPanels(merged),
-            }),
-          });
+          const now = Date.now();
+          // Prevent duplicate processing of the same data
+          // Skip if we processed this exact data less than 100ms ago
+          if (this.lastProcessedData === frame && now - this.lastProcessedTime < 100) {
+            return;
+          }
+          
+          // Track this data as processed
+          this.lastProcessedData = frame;
+          this.lastProcessedTime = now;
+          
+          try {
+            const response = JSON.parse(frame) as TraceSearchMetadata[];
+            const merged = mergeTraces(response);
+            merged.children.sort((a, b) => countSpans(b) - countSpans(a));
+            
+            if (state.data?.state === LoadingState.Streaming) {
+              // Clear any pending update
+              if (this.updateTimeoutId) {
+                clearTimeout(this.updateTimeoutId);
+              }
+              
+              // Set a timeout to apply the update after the debounce time
+              this.updateTimeoutId = setTimeout(() => {
+                this.updatePanels(merged);
+                this.updateTimeoutId = null;
+              }, STREAMING_DEBOUNCE_TIME_MS);
+              
+              return;
+            }
+            
+            // For done state, update immediately
+            this.updatePanels(merged);
+          } catch (error) {
+            console.error('Error processing trace data:', error);
```

Comment:
Should we show the error to the user or we intentionally skip it silently? Would it make sense to push it at least to faro?

Reply:
I've removed for now as it's not necessary for the PR.